doc/adr/0007-api-service.md

```diff
@@ -0,0 +1,106 @@
+
+To illustrate, you could say that the `wego` command line could tell the wego-api "Add this application" and it, in turn, would tell Kubernetes "Apply these manifest files. "
+
+For security reasons, permissions will be implemented through RBAC so that the wego-api only has access to the resources it needs.
+
+With this proposal, the use of CLI options would still be required, however, it would be an intermediate step to eventually do
+a 100% migration into the cluster.
+
+Specifically, the proposed changes are as follows:
+
+### Create wego-api docker image and update authentication method.
+
+For the wego-api to run within the cluster, a docker image will need to exist that can run it.
+
+The part that needs to be run in this image is the HTTP server which is initialized in the `wego ui run` command.
+
+To continue communicating with the Kubernetes API, it will be required to change the authentication method to one that allows communication from within the cluster called in-cluster client configuration.
+
+Kubectl and flux will be added to this docker image as well.
+
+### Generate RBAC-based permissions to the wego-api and apply them when installing the GitOps runtime.
+
+Since the wego-api needs to access resources within the cluster, RBAC permissions will be applied so that there are no security gaps. In consideration that the functionality of the wego-api will be specified only in actions within the namespace where the GitOps runtime is executed, the Role and RoleBinding resources will be used. Since the scope of this type of resource is by namespace. 
+
+The resources that will be required are:
+
+- ServiceAccount
+- Deployment
+- Role
+- RoleBinding
+- Service
+
+The specific permissions for the Role resource will be:
+
+```
+resources: ["apps.wego.weave.works"]
+verbs: ["get","create","list"]
+
+resources: ["secret"]
+verbs: ["get","create"]
+
+resources: ["kustomizations.kustomize.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+
+resources: ["helmreleases.helm.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+
+resources: ["helmrepositories.source.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+```
+
+These resources will be applied when installing the GitOps engine.
+
+### Enable access to the wego-api from the client-side.
+
+To enable access to the `wego-api` from the client machine, we will use `kubectl` by running this command: `kubectl -n NAMESPACE port-forward deployment/wego-api :8283`. An example of the output would be: 
+```
+Forwarding from 127.0.0.1:63701 -> 8283
+Forwarding from [::1]:63701 -> 8283
+```
+
+The path to access `wego-api` will be `127.0.0.1:63701` 
+
+This process of exposing `wego-api` will be executed for each command that is executed from CLI. With the exception of the `wego ui run` command that will keep `wego-api` exposure execution in the background for the duration of that command session, so that the UI can access the resource.
```

Comment:
Please elaborate on the `wego ui run` command.  e.g., does it open the port to the service, then open a browser to a certain address that loads the SPA from the API service?